scripts/install/RegisterItemRelationsService.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2025 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\scripts\install;
+
+use oat\oatbox\extension\InstallAction;
+use oat\tao\model\resources\relation\service\ResourceRelationServiceProxy;
+use oat\tao\model\search\SearchProxy;
+use oat\taoAdvancedSearch\model\Resource\Service\ItemRelationsService;
+
+class RegisterItemRelationsService extends InstallAction
+{
+    public function __invoke($params)
+    {
+        if ($this->getServiceManager()->get(SearchProxy::class)->supportCustomIndex()) {
+            $resourceRelationService = $this->getServiceManager()->get(ResourceRelationServiceProxy::SERVICE_ID);
+            $services = $resourceRelationService->getOption(ResourceRelationServiceProxy::OPTION_SERVICES);
+
+            if (!isset($services['test'])) {
+                $services['test'] = [];
+            }
+
+            if (is_array($services['test'])) {
+                $services['test'][] = ItemRelationsService::class;
+            }
+
+            $resourceRelationService->setOption(ResourceRelationServiceProxy::OPTION_SERVICES, $services);
+
+            $this->getServiceManager()->register(
+                ResourceRelationServiceProxy::SERVICE_ID, $resourceRelationService
+            );
+        }
+    }
```

Comment:
You do not need all this logic, please simply use `$resourceRelationService->addService('test', ItemRelationsService::class);`

```suggestion
        if ($this->getServiceManager()->get(SearchProxy::class)->supportCustomIndex()) {
            $resourceRelationService = $this->getServiceManager()->get(ResourceRelationServiceProxy::SERVICE_ID);
            $resourceRelationService->addService('test', ItemRelationsService::class);

            $this->getServiceManager()->register(
                ResourceRelationServiceProxy::SERVICE_ID, $resourceRelationService
            );
        }
    }
```